src/regex/match_data.cr

```diff
@@ -219,11 +220,15 @@ class Regex
     end
 
     private def check_index_out_of_bounds(index)
-      raise IndexError.new unless valid_group?(index)
+      raise_invalid_group_index(index) unless valid_group?(index)
     end
 
     private def valid_group?(index)
       index <= @size
     end
+
+    private def raise_invalid_group_index(n)
```

Comment:
A small suggestion: following the `index` argument name from the callers, perhaps the method signature reads better as `raise_invalid_group(index)`?